velox/functions/lib/Re2Functions.cpp

```diff
@@ -381,6 +406,235 @@ class Re2SearchAndExtract final : public VectorFunction {
   const bool emptyNoMatch_;
 };
 
+class Re2ReplaceConstantPattern final : public VectorFunction {
+ public:
+  /**
+   * @brief Constructor for the Re2ReplaceConstantPattern class.
+   *
+   * This class provides a method to replace occurrences of a pattern in a
+   * string starting from a given position.
+   *
+   * @param basePattern The regular expression pattern to search for in the
+   * input string.
+   */
+  explicit Re2ReplaceConstantPattern(StringView basePattern)
+      : basePattern_(basePattern.getString()) {}
```

Comment:
maybe rename `basePattern_` to `re_` to keep consistent with existing Re2Functions?